File: app/src/main/java/com/example/bhagavadgita/data/model/Verses.java
package com.example.bhagavadgita.data.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;

public class Verses {

    @SerializedName("id")
    @Expose
    public int id;
    @SerializedName("verse_number")
    @Expose
    public int verse_number;
    @SerializedName("chapter_number")
    @Expose
    public int chapter_number;
    @SerializedName("slug")
    @Expose
    public String slug;
    @SerializedName("text")
    @Expose
    public String text;
    @SerializedName("transliteration")
    @Expose
    public String transliteration;
    @SerializedName("word_meanings")
    @Expose
    public String word_meanings;
    @SerializedName("translations")
    @Expose
    public ArrayList<Translation> translations;
    @SerializedName("commentaries")
    @Expose
    public ArrayList<Commentary> commentaries;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getVerse_number() {
        return verse_number;
    }

    public void setVerse_number(int verse_number) {
        this.verse_number = verse_number;
    }

    public int getChapter_number() {
        return chapter_number;
    }

    public void setChapter_number(int chapter_number) {
        this.chapter_number = chapter_number;
    }

    public String getSlug() {
        return slug;
    }

    public void setSlug(String slug) {
        this.slug = slug;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public String getTransliteration() {
        return transliteration;
    }

    public void setTransliteration(String transliteration) {
        this.transliteration = transliteration;
    }

    public String getWord_meanings() {
        return word_meanings;
    }

    public void setWord_meanings(String word_meanings) {
        this.word_meanings = word_meanings;
    }

    public ArrayList<Translation> getTranslations() {
        return translations;
    }

    public void setTranslations(ArrayList<Translation> translations) {
        this.translations = translations;
    }

    public ArrayList<Commentary> getCommentaries() {
        return commentaries;
    }

    public void setCommentaries(ArrayList<Commentary> commentaries) {
        this.commentaries = commentaries;
    }
}

File: app/src/main/java/com/example/bhagavadgita/data/model/Verses.kt
package com.example.bhagavadgita.data.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.TypeConverters
import com.example.bhagavadgita.utils.CommentaryListConverter
import com.example.bhagavadgita.utils.TranslationListConverter
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName


@Entity(tableName = "verses")
data class Verses(
    @PrimaryKey
    @ColumnInfo(name = "id")
    @SerializedName("id")
    @Expose
    var id: Int,

    @ColumnInfo(name = "verse_number")
    @SerializedName("verse_number")
    @Expose
    var verseNumber: Int,

    @ColumnInfo(name = "chapter_number")
    @SerializedName("chapter_number")
    @Expose
    var chapterNumber: Int,

    @ColumnInfo(name = "slug")
    @SerializedName("slug")
    @Expose
    var slug: String,

    @ColumnInfo(name = "text")
    @SerializedName("text")
    @Expose
    var text: String,

    @ColumnInfo(name = "transliteration")
    @SerializedName("transliteration")
    @Expose
    var transliteration: String,

    @ColumnInfo(name = "word_meanings")
    @SerializedName("word_meanings")
    @Expose
    var wordMeanings: String,

    @ColumnInfo(name = "translations")
    @SerializedName("translations")
    @Expose
    @TypeConverters(TranslationListConverter::class)
    var translations: ArrayList<Translation>,

    @ColumnInfo(name = "commentaries")
    @SerializedName("commentaries")
    @Expose
    @TypeConverters(CommentaryListConverter::class)
    var commentaries: ArrayList<Commentary>
){
    constructor() : this(
        id = 0,
        verseNumber = 0,
        chapterNumber = 0,
        slug = "",
        text = "",
        transliteration = "",
        wordMeanings = "",
        translations = arrayListOf(),
        commentaries = arrayListOf()
    )
}

